### ISSUE TITLE
[BUG] More Activities - Test Your Smarts Fails to Complete (Solution Found)

### ISSUE THREAD

**Issue by AlexyGalaxy** [charlesbel/Microsoft-Rewards-Farmer#142]
I am not sure why but test your smarts fails to complete under More Activities is not completing with the current version of the script. Currently the page opens, quiz starts, first option is selected, then the page waits and quits. The quiz has 10 questions in total and has 3 options per question.

the script is having trouble with line 631
numberOfQuestions = max([int(s) for s in counter.split() if s.isdigit()])

I am going to look into this however if someone else knows of a solution please post it here.

**Comment by AlexyGalaxy** [comment]
I just made a fix to this issue. Delete code from line 630-636 and replace it with the code below.
```
try :
        counter = str(browser.find_element(By.XPATH, '//*[@id="QuestionPane0"]/div[2]').get_attribute('innerHTML'))[:-1][1:]
        print('counter:'+counter)
        numberOfQuestions = (counter.split(" of ")[1])
        try:
            for question in range(int(numberOfQuestions)):
                browser.execute_script('document.evaluate("//*[@id=\'QuestionPane' + str(question) + '\']/div[1]/div[2]/a[' + str(random.randint(1, 3)) + ']/div", document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue.click()')
                time.sleep(5)
                try:   
                    browser.find_element(By.XPATH, '//*[@id="AnswerPane' + str(question) + '"]/div[1]/div[2]/div[4]/a/div/span/input').click()
                    time.sleep(2)
                except:
                    pass
                browser.find_element(By.XPATH, '//*[@id="nextQuestionbtn' + str(question) + '"]').click() #should fix more promo quiz
                time.sleep(5)
        except :
            prRed('\n[ERROR] completeMorePromotionABC failed')
```

**Comment by AlexyGalaxy** [comment]
I forked this project and uploaded my current working version that will complete all searches and punch cards. Also made the README even more simple to understand.

https://github.com/AlexyGalaxy/Microsoft-Rewards-Farmer

**Comment by kingnab** [comment]
i keep getting errors like
[ERROR] An Error has Occured While Trying to Complete More Promotion Search.
←[00m
←[91m
[ERROR] An Error has Occured While Trying to Write Err.
←[00m
←[91m
[ERROR] An Error has Occured While Trying to Complete More Promotion Quiz.
←[00m
←[91m
[ERROR] An Error has Occured While Trying to Write Err.
←[00m
←[91m
[ERROR] An Error has Occured While Trying to Complete More Promotion Quiz.
←[00m
←[91m
[ERROR] An Error has Occured While Trying to Write Err.
←[00m

**Comment by kingnab** [comment]
im also using ur files @AlexyGalaxy  pls help

**Comment by AlexyGalaxy** [comment]
You dont seem to have the error logger set up correctly. 

-First make sure you are up to date with your MS Farmer
-Set up both RewardsErr and RewardsLog to output the .txt to anywhere on your computer by chaing the YOURNAMEHERE to your PC's Username

There shouldnt be an issue with completing promos, however you can disable headless mode by following the steps in the ReadMe which will allow you to watch the program work. Watch it and when you notice the error manually open the rewards dashboard outside of the script in a normal browser and take a screen shot of the pannel so I can provide additional assistance.

**Comment by charlesbel** [comment]
Fixed with v3.0 #188
